rust/datafusion/src/dataframe.rs

```diff
@@ -0,0 +1,168 @@
+    ///
+    /// let mut ctx = ExecutionContext::new();
+    /// let df = ctx.read_csv("tests/example.csv", CsvReadOptions::new()).unwrap();
+    /// let df = df.select(vec![col("a").multiply(col("b")), col("c")]).unwrap();
+    /// ```
+    fn select(&self, expr: Vec<Expr>) -> Result<Arc<dyn DataFrame>>;
+
+    /// Filter a DataFrame to only include rows that match the specified filter expression.
+    ///
+    /// ```
+    /// use datafusion::ExecutionContext;
+    /// use datafusion::execution::physical_plan::csv::CsvReadOptions;
+    /// use datafusion::logicalplan::col;
+    ///
+    /// let mut ctx = ExecutionContext::new();
+    /// let df = ctx.read_csv("tests/example.csv", CsvReadOptions::new()).unwrap();
+    /// let df = df.filter(col("a").lt_eq(col("b"))).unwrap();
+    /// ```
+    fn filter(&self, expr: Expr) -> Result<Arc<dyn DataFrame>>;
+
+    /// Perform an aggregate query with optional grouping expressions.
+    ///
+    /// ```
+    /// use datafusion::ExecutionContext;
+    /// use datafusion::execution::physical_plan::csv::CsvReadOptions;
+    /// use datafusion::logicalplan::col;
+    ///
+    /// let mut ctx = ExecutionContext::new();
+    /// let df = ctx.read_csv("tests/example.csv", CsvReadOptions::new()).unwrap();
+    ///
+    /// // The following use is the equivalent of "SELECT MIN(b) GROUP BY a"
+    /// let _ = df.aggregate(vec![col("a")], vec![df.min(col("b")).unwrap()]).unwrap();
+    ///
+    /// // The following use is the equivalent of "SELECT MIN(b)"
+    /// let _ = df.aggregate(vec![], vec![df.min(col("b")).unwrap()]).unwrap();
+    /// ```
+    fn aggregate(
+        &self,
+        group_expr: Vec<Expr>,
+        aggr_expr: Vec<Expr>,
+    ) -> Result<Arc<dyn DataFrame>>;
+
+    /// limit the number of rows returned from this DataFrame.
+    ///
+    /// ```
+    /// use datafusion::ExecutionContext;
+    /// use datafusion::execution::physical_plan::csv::CsvReadOptions;
+    /// use datafusion::logicalplan::col;
+    ///
+    /// let mut ctx = ExecutionContext::new();
+    /// let df = ctx.read_csv("tests/example.csv", CsvReadOptions::new()).unwrap();
+    /// let df = df.limit(100).unwrap();
+    /// ```
+    fn limit(&self, n: usize) -> Result<Arc<dyn DataFrame>>;
+
+    /// Executes this DataFrame and collects all results into a vector of RecordBatch.
+    ///
+    /// ```
+    /// use datafusion::ExecutionContext;
+    /// use datafusion::execution::physical_plan::csv::CsvReadOptions;
+    /// use datafusion::logicalplan::col;
+    ///
+    /// let mut ctx = ExecutionContext::new();
+    /// let df = ctx.read_csv("tests/example.csv", CsvReadOptions::new()).unwrap();
+    /// let batches = df.collect(4096).unwrap();
+    /// ```
+    fn collect(&self, batch_size: usize) -> Result<Vec<RecordBatch>>;
+
+    /// Returns the schema describing the output of this DataFrame in terms of columns returned,
+    /// where each column has a name, data type, and nullability attribute.
+
```

Comment:
I am not sure if this won't break descriptions (add `    ///`?). If not, ignore this comment.